## US Gun Deaths Data Set

[Original article by FiveThirtyEight about Guns](http://fivethirtyeight.com/features/gun-deaths/)

The data set contains cleaned gun-death data from the CDC for 2012-2014.

### Assignment

- Import the csv
- Read it into a list
- Preview the first 5 entries

In [1]:
# opens the guns.csv file and processes using csv.reader, converts the dataset into a list of lists, prints first
# 5 entries

import csv
f = open("guns.csv", "r")
deaths = csv.reader(f)
list_of_rows = []
for row in deaths:
    list_of_rows.append(row)
print(list_of_rows[0:5])

[['', 'year', 'month', 'intent', 'police', 'sex', 'age', 'race', 'hispanic', 'place', 'education'], ['1', '2012', '01', 'Suicide', '0', 'M', '34', 'Asian/Pacific Islander', '100', 'Home', '4'], ['2', '2012', '01', 'Suicide', '0', 'F', '21', 'White', '100', 'Street', '3'], ['3', '2012', '01', 'Suicide', '0', 'M', '60', 'White', '100', 'Other specified', '4'], ['4', '2012', '02', 'Suicide', '0', 'M', '64', 'White', '100', 'Home', '4']]


### Assignment

- Remove the header row from the list of lists
- Save it to a separate list

In [2]:
# removes header row from the list of lists and save it as a seperate list
header_row = list_of_rows.pop(0)
print(header_row)

['', 'year', 'month', 'intent', 'police', 'sex', 'age', 'race', 'hispanic', 'place', 'education']


### Assignment

- Count the number of gun deaths by year
    - It may help to do a list comprehension to get the years
    - Iterate over the years with a dictionary to keep count
    
    

In [3]:
# creates a dictionary that summerizes the number of gun deaths per year as reported in the dataset
from collections import defaultdict
gun_deaths_by_year = defaultdict(int)
for ls in list_of_rows:
    gun_deaths_by_year[ls[1]] += 1
print(gun_deaths_by_year) 


defaultdict(<class 'int'>, {'2012': 33563, '2013': 33636, '2014': 33599})


### Assignment

- Import the datetime library
- Create a new list called "dates" with values from the data (set all the day values to 1)    
- Count they number of gun deaths by month and year



In [5]:
# creates a dictionary that summerizes the number of gun deaths per month as reported in the dataset.  Date format is 
# (month, year).
import datetime as dt
date1 = '2012-01-01'
date2 = '2015-1-01'
start = dt.datetime.strptime(date1, '%Y-%m-%d')
end = dt.datetime.strptime(date2, '%Y-%m-%d')
step = dt.timedelta(days=31)
dates = []
while start <= end:
    #dates.append(start.date())
    dates.append(start.timetuple())
    start += step

gun_deaths_by_date = defaultdict(int)    
for i in range(len(dates)):
    date_key = dates[i][1], dates[i][0]
    for ls in list_of_rows:
        if int(ls[1]) == int(dates[i][0]) and int(ls[2]) == int(dates[i][1]):
            gun_deaths_by_date[date_key] += 1

print(gun_deaths_by_date)            




defaultdict(<class 'int'>, {(1, 2012): 2758, (2, 2012): 2357, (3, 2012): 2743, (4, 2012): 2795, (5, 2012): 2999, (6, 2012): 2826, (7, 2012): 3026, (8, 2012): 2954, (9, 2012): 2852, (10, 2012): 2733, (11, 2012): 2729, (12, 2012): 2791, (1, 2013): 2864, (2, 2013): 2375, (3, 2013): 2862, (4, 2013): 2798, (5, 2013): 2806, (6, 2013): 2920, (7, 2013): 3079, (8, 2013): 2859, (9, 2013): 2742, (10, 2013): 2808, (11, 2013): 2758, (12, 2013): 2765, (1, 2014): 2651, (2, 2014): 2361, (3, 2014): 2684, (4, 2014): 2862, (5, 2014): 2864, (6, 2014): 2931, (7, 2014): 2884, (8, 2014): 2970, (9, 2014): 2914, (10, 2014): 2865, (11, 2014): 2756, (12, 2014): 2857})


### Assignment

- Find the number of gun deaths by Sex
- Find the number of gun deaths by Race
- How does this compare to the overall population in the US?

In [6]:
# creates a dictionary that summerizes the number of gun deaths per sex as reported in the dataset
gun_deaths_by_sex = defaultdict(int)
for ls in list_of_rows:
    gun_deaths_by_sex[ls[5]] += 1
print(gun_deaths_by_sex) 

# creates a dictionary that summerizes the number of gun deaths per race as reported in the dataset
gun_deaths_by_race = defaultdict(int)
for ls in list_of_rows:
    gun_deaths_by_race[ls[7]] += 1
print(gun_deaths_by_race)


defaultdict(<class 'int'>, {'M': 86349, 'F': 14449})
defaultdict(<class 'int'>, {'Asian/Pacific Islander': 1326, 'White': 66237, 'Native American/Native Alaskan': 917, 'Black': 23296, 'Hispanic': 9022})


### Assignment

- Reuse the data structure counting deaths by race
- Use the dictionary below that has the actual population of each race
- Compute the rates of gun deaths per race per 100,000 people

mapping = {
    "Asian/Pacific Islander": 15159516 + 674625,
    "Native American/Native Alaskan": 3739506,
    "Black": 40250635,
    "Hispanic": 44618105,
    "White": 197318956
}

In [7]:
# computes the rates of gun deaths per race per 100,000 people
mapping = {'Asian/Pacific Islander': 15834141, 'Native American/Native Alaskan': 3739506, 'Black': 40250635, 
           'Hispanic': 44618105, 'White': 197318956}
print("Gun deaths by race per 100,000 people")
for key in gun_deaths_by_race:
    per_100000 = round((gun_deaths_by_race[key]/mapping[key]) * 100000, 2)
    print(key, per_100000)


Gun deaths by race per 100,000 people
Asian/Pacific Islander 8.37
White 33.57
Native American/Native Alaskan 24.52
Black 57.88
Hispanic 20.22


### Assignment

You may not know this, but over half of all gun deaths are suicide.

- Redo the computation of rates of gun deaths per race per 100,000 people
- This time only count those that are "Homicide"
- How are these different than the previous calculation?


In [8]:
# computes the rates of homicide gun deaths per race per 100,000 people
homicide_gun_deaths_by_race = defaultdict(int)
for ls in list_of_rows:
    if ls[3] == 'Homicide':
        homicide_gun_deaths_by_race[ls[7]] += 1

print("Homicide gun deaths by race per 100,000 people")
for key in homicide_gun_deaths_by_race:
    per_100000 = round((homicide_gun_deaths_by_race[key]/mapping[key]) * 100000, 2)
    print(key, per_100000)


Homicide gun deaths by race per 100,000 people
White 4.64
Asian/Pacific Islander 3.53
Black 48.47
Native American/Native Alaskan 8.72
Hispanic 12.63


After calculating the total number of gun deaths per 100,000 people and the subset of those gun deaths that occured by homicide, it is clear that when compared to other races, black people are more likey to be killed by a gun in general (57.88 per 100,000) and as a result of homicide (48.47 per 100,000).  In fact they are 4 times more likely to be killed by a gun homicide than Hispanics, which have second highest rate of gun homicides (12.63 per 1000,000).  While whites have the second highest frequency of gun deaths per 100,000 (33.57) they have the second lowest frequency of gun homicides (4.67) which gives them the lowest percentage of total gun deaths that are homicides.  Asian/Pacific Islander have the lowest gun deaths and gun homicides per 100,000 of any racial group at 8.37 and 3.53 respectively.    
                        